packages/astro/src/astro-components/interactive/OrganizationList.astro

```diff
@@ -0,0 +1,36 @@
+---
+import type { OrganizationListProps } from "@clerk/types";
+type Props = OrganizationListProps
+
+import { customAlphabet, urlAlphabet } from "nanoid";
+
+const safeId = customAlphabet(urlAlphabet, 10)();
+
+const props = {
+  ...Astro.props,
+}
+---
+
+<div id={`clerk-organization-list-${safeId}`}></div>
+
+<script is:inline define:vars={{ props: Astro.props, safeId }}>
```

Comment:
The `safeId` and `setOrCreatePropMap` are repeated in multiple Astro components. Can we consolidate that into shared utils?

Reply:
We will resolve this in a follow up PR, i've documented it